data/pc/latest/proto.yml

```diff
@@ -2155,71 +2156,70 @@
-                     ingredient: SlotDisplay
-                     fuel: SlotDisplay
-                     result: SlotDisplay
-                     craftingStation: SlotDisplay
-                     duration: varint
-                     experience: f32
-                  if stonecutter:
-                     ingredient: SlotDisplay
-                     result: SlotDisplay
-                     craftingStation: SlotDisplay
-                  if smithing:
-                     template: SlotDisplay
-                     base: SlotDisplay
-                     addition: SlotDisplay
-                     result: SlotDisplay
-                     craftingStation: SlotDisplay
-            group?: varint
-            # This ID corresponds to the "recipe_book_category" registry
-            category: varint =>
-            - crafting_building_blocks
-            - crafting_redstone
-            - crafting_equipment
-            - crafting_misc
-            - furnace_food
-            - furnace_blocks
-            - furnace_misc
-            - blast_furnace_blocks
-            - blast_furnace_misc
-            - smoker_food
+         recipeId: varint
+         display:
+            type: varint =>
+            - crafting_shapeless
+            - crafting_shaped
+            - furnace
             - stonecutter
             - smithing
-            - campfire
-            craftingRequirements?: ["registryEntryHolderSet", {
+            data: type ?
+               if crafting_shapeless:
+                  ingredients: SlotDisplay[]varint
+                  result: SlotDisplay
+                  craftingStation: SlotDisplay
+               if crafting_shaped:
+                  width: varint
+                  height: varint
+                  ingredients: SlotDisplay[]varint
+                  result: SlotDisplay
+                  craftingStation: SlotDisplay
+               if furnace:
+                  ingredient: SlotDisplay
+                  fuel: SlotDisplay
+                  result: SlotDisplay
+                  craftingStation: SlotDisplay
+                  duration: varint
+                  experience: f32
+               if stonecutter:
+                  ingredient: SlotDisplay
+                  result: SlotDisplay
+                  craftingStation: SlotDisplay
+               if smithing:
+                  template: SlotDisplay
+                  base: SlotDisplay
+                  addition: SlotDisplay
+                  result: SlotDisplay
+                  craftingStation: SlotDisplay
+         groupId: varint
+         categoryId: varint =>
+         - crafting_building_blocks
+         - crafting_redstone
+         - crafting_equipment
+         - crafting_misc
+         - furnace_food
+         - furnace_blocks
+         - furnace_misc
+         - blast_furnace_blocks
+         - blast_furnace_misc
+         - smoker_food
+         - stonecutter
+         - smithing
+         - campfire
+         hasIngredients: bool
```

Comment:
This is unneeded, `craftingRequirements?:` does the same, `?` means it's an optional field